src/rust/bitbox02/src/delay.rs

```diff
@@ -0,0 +1,151 @@
+// Copyright 2025 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use core::pin::Pin;
+use core::task::{Context, Poll};
+
+enum DelayState {
+    Unstarted,
+    Running,
+    Done,
+}
+
+#[cfg(not(any(feature = "testing", feature = "c-unit-testing")))]
+struct DelayInner {
+    bitbox02_delay: bitbox02_sys::delay_t,
+}
+
+#[cfg(any(feature = "testing", feature = "c-unit-testing"))]
+struct DelayInner {
+    thread_handle: Option<std::thread::JoinHandle<()>>,
+    done: std::sync::Arc<std::sync::atomic::AtomicBool>,
```

Comment:
why is this needed when the DelayState also has a Done arm? Can't you just move the state to `Done` directly?